sdk/src/nonce_account.rs

```diff
@@ -20,14 +22,13 @@ pub fn create_account(lamports: u64) -> RefCell<AccountSharedData> {
     )
 }
 
-// TODO: Consider changing argument from Hash to DurableNonce.
-pub fn verify_nonce_account(acc: &AccountSharedData, hash: &Hash) -> bool {
```

Comment:
> What makes it "necessary" to remove this function and change the call sites?

Definitely not "necessary" to remove this function

> Change this function to Option<State> to return deserialized State on success so that the change to the call sites is bare minimum.

This is pretty much the line of reasoning I went through when implementing this change. After changing this to return `Option<Data>`, I found that other than `Bank::check_message_for_nonce`, all other call sites only needed to check whether the nonce was advanced or not.

Lastly, I feel that runtime checks for nonce account verification belong in the runtime not the sdk

Reply:
Never mind, I see that we can keep that method and still have the runtime signer check be separate. I've adapted your feedback now and it's ready for another pass